volue_insight_timeseries/util.py

```diff
@@ -26,29 +27,38 @@
 
 # Frequency mapping from TS to Pandas
 _TS_FREQ_TABLE = {
-    'Y': 'AS',
+    'Y': 'YS',
     'S': '2QS',
     'Q': 'QS',
     'M': 'MS',
     'W': 'W-MON',
-    'H12': '12H',
-    'H6': '6H',
-    'H3': '3H',
-    'MIN30': '30T',
-    'MIN15': '15T',
-    'MIN5': '5T',
-    'MIN': 'T',
+    'H12': '12h',
+    'H6': '6h',
+    'H3': '3h',
+    'MIN30': '30min',
+    'MIN15': '15min',
+    'MIN5': '5min',
+    'MIN': 'min',
 }
-# Mapping from Pandas to TS is built from map above, with some additions
+
+# Mapping from Pandas to TS is built from map above,
+# with some additions to support older versions of pandas
 _PANDAS_FREQ_TABLE = {
+    'YS-JAN': 'Y',
     'AS-JAN': 'Y',
-    'YS': 'Y',
+    'AS': 'Y',
     '2QS-JAN': 'S',
     'QS-JAN': 'Q',
-    'min': 'MIN',
+    '12H': 'H12',
+    '6H': 'H6',
+    '3H': 'H3',
+    '30T': 'MIN30',
+    '15T': 'MIN15',
+    '5T': 'MIN5',
+    'T': 'MIN',
 }
 for k, v in _TS_FREQ_TABLE.items():
-    _PANDAS_FREQ_TABLE[v] = k
+    _PANDAS_FREQ_TABLE[v.upper()] = k
```

Comment:
Are we sure that's necessary? I think the values in the dictionary should be as is.

Reply:
It's necessary given the other code parts, see: https://github.com/volueinsight/volue-insight-timeseries/blob/c6b645ece566af9c2e136aac691172c922d75a79/volue_insight_timeseries/util.py#L167

Removing the `upper()` in `_rev_map_freq` would introduce a potentially breaking change that could go undetected, so I wanted to limit the introduction of potential issues. This may be something for a later update/refactoring, maybe with a prior deprecation warning.